minecraft/bukkit/src/main/java/dev/triumphteam/cmd/bukkit/BukkitCommandProcessor.java

```diff
@@ -87,19 +88,17 @@ protected BukkitSubCommand<S> createSubCommand(
         return new BukkitSubCommand<>(processor, getName(), executionProvider);
     }
 
-    static CommandPermission createPermission(
-            @NotNull final String parentNode,
-            @NotNull final String node,
+    static List<CommandPermission> createPermissions(
+            @NotNull final String[] parentNode, // TODO determine usage?
+            @NotNull final String[] node,
             @NotNull final String description,
             @NotNull final PermissionDefault permissionDefault
     ) {
-        final StringBuilder permissionBuilder = new StringBuilder();
-        if (!parentNode.isEmpty()) {
-            permissionBuilder.append(parentNode);
-            if (!node.isEmpty()) permissionBuilder.append(".");
+        List<CommandPermission> permissions = new ArrayList<>();
```

Comment:
This right here is where it needs a bit more work, currently it's removing a feature.
Previously if you have:
```
@Permission("test.permission")
class MyCommand {
    @Permission("sub")
    void execute() {}
}
```
The final result would be `test.permission.sub`, so basically it concatenates the parent node to the current node.
So in theory when having multiple permissions let's say parent nodes are `parent.test` and `parent.alias`, and the current node is `subcommand`, then you should have two permissions, `parent.test.subcommand` and `parent.alias.subcommand`.

Reply:
Ah, gotcha. I was trying to figure out what that `parentNode` variable was for and I couldn't figure it out for the life of me. I will implement these requested fixes today.

Reply:
That's totally fine, feel free to ask me whenever there is something you don't understand and I'll help you out ;p